packages/mobile/src/send/useInviteRewardLimitDialog.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useSelector } from 'react-redux'
+import Dialog from 'src/components/Dialog'
+import { MAX_NUM_INVITES_WITH_REWARDS } from 'src/config'
+import { getAddressFromPhoneNumber } from 'src/identity/contactMapping'
+import {
+  e164NumberToAddressSelector,
+  secureSendPhoneNumberMappingSelector,
+} from 'src/identity/selectors'
+import { Recipient } from 'src/recipients/recipient'
+import { inviteRewardsActiveSelector, invitesWithRewardsSentSelector } from 'src/send/selectors'
+
+const useInviteRewardLimitDialog = (onNavigateSendAmount: (recipient: Recipient) => void) => {
+  const [isVisible, setIsVisible] = useState(false)
+  const [selectedRecipient, setSelectedRecipient] = useState<Recipient | null>(null)
+
+  const invitesWithRewardsSent = useSelector(invitesWithRewardsSentSelector)
+  const inviteRewardsEnabled = useSelector(inviteRewardsActiveSelector)
+  const secureSendPhoneNumberMapping = useSelector(secureSendPhoneNumberMappingSelector)
+  const e164NumberToAddress = useSelector(e164NumberToAddressSelector)
+
+  const { t } = useTranslation()
+
+  const handleClose = () => {
+    setIsVisible(false)
+  }
+
+  const handleInvite = () => {
+    if (selectedRecipient) {
+      setIsVisible(false)
+      onNavigateSendAmount(selectedRecipient)
+    }
+  }
+
+  const onProceedSelectRecipient = (recipient: Recipient) => {
+    if (inviteRewardsEnabled && !recipient.address && recipient.e164PhoneNumber) {
+      const recipientAddress = getAddressFromPhoneNumber(
+        recipient.e164PhoneNumber,
+        e164NumberToAddress,
+        secureSendPhoneNumberMapping,
+        undefined
+      )
+      const isInvite = !recipientAddress
+      const hasReachedRewardLimit = invitesWithRewardsSent === MAX_NUM_INVITES_WITH_REWARDS
```

Comment:
ah, i thought about this but then the user would be warned every time after the 5th invite. we could introduce a new redux variable to prevent this of course, but that felt heavy. i think after this program finishes, we should introduce a new redux migration that resets this count to 0 when the user updates their app so that we can start the count fresh for a new program